### PR TITLE
Use generated ProwJob client in sinker

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Biting off small bits of the pie first.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @cjwagner @fejta 
/cc @BenTheElder @krzyzacy

### CODE REVIEW

File: prow/cmd/sinker/main.go [hunk 1]
```diff
@@ -91,10 +102,10 @@ func main() {
 		kubeClients[alias] = kubeClient(client)
 	}
 	c := controller{
-		logger:      logrus.NewEntry(logrus.StandardLogger()),
-		kc:          kc,
-		pkcs:        kubeClients,
-		configAgent: configAgent,
+		logger:        logrus.NewEntry(logrus.StandardLogger()),
+		prowJobClient: pjclient.ProwV1().ProwJobs(configAgent.Config().ProwJobNamespace),
+		pkcs:          kubeClients,
```
Comment (krzyzacy) [inline prow/cmd/sinker/main.go]: Just curious if we can also use the real client for pods instead of our kube.Client?

Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: It's a follow up :)

Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: After the other massive PR I want to make small changes and not rebase hell

Comment (fejta) [reply prow/cmd/sinker/main.go]: Add a TODO to remove this?

File: prow/cmd/sinker/main.go [hunk 2]
```diff
@@ -68,6 +69,16 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 
+	clusterConfig, err := rest.InClusterConfig()
```
Comment (fejta) [inline prow/cmd/sinker/main.go]: I think we should continue supporting the existing auth methods until everything has migrated.

We can just encapsulate this into something reusable:

https://github.com/kubernetes/test-infra/blob/aa8d6d1c02a9deb17f3c631a4ac1dc87885fbdb7/prow/cmd/build/main.go#L157-L194

**Comment by fejta** [comment]
(testing need release)

File: prow/cmd/sinker/main.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: I have the second pull already made. No need.

File: prow/cmd/sinker/main.go (hunk 2 above)
Comment (stevekuznetsov) [reply prow/cmd/sinker/main.go]: The PJ client was always in cluster, though. I don't understand you.

**Comment by stevekuznetsov** [comment]
@fejta this changes nothing about the method that is supported. PJ client was always simply the in cluster configuration and this commit just changes it to use the upstream types.

/hold cancel

**Pull request merged**